internal/testrunner/runners/pipeline/test_result.go

```diff
@@ -73,6 +75,81 @@ func compareResults(testCasePath string, config *testConfig, result *testResult)
 	return nil
 }
 
+// diffUlite implements a unified diff-like rendering with u lines of context.
+// It differs from a complete unified diff in that it does not provide the length
+// of the chunk, so it cannot be used to generate patches, but can be used for
+// human inspection.
+func diffUlite(a, b string, u int) string {
```

Comment:
Could you please make sure that there isn't any Go library, that can deliver the same functionality? I would rather follow that approach than implementing it on your own.

Maybe [go-diff](https://github.com/sergi/go-diff)?

Reply:
I did a survey first. go-diff is much more complicated than is needed here. My initial intention was to use go-cmp, but it is too focused on Go use and though this is a Go program, integrations should not assume users have any Go coding experience. The second package I looked at was indeed go-diff, but it is massively over`engineered for what is needed with an overcomplicated API.